docs/method-monitoring.asciidoc

```diff
@@ -0,0 +1,282 @@
+include::method-monitoring.asciidoc[tag=combine-api-annotations]
+
+//********************************************
+[[method-annotations]]
+=== Annotations
+
+The <<api-annotation,annotation API>> allows you to place annotations on top of methods to automatically create spans for them.
+This method of creating spans is easier, more robust, and typically more performant than using the API;
+there’s nothing you can do wrong like forgetting to end a span or close a scope.
+
+Annotations are less flexible when used on their own, but can be combined with the span API for added flexibility.
+
+[float]
+==== How-to create spans with the annotations API
+
+Here's an example that uses the <<api-capture-span,`@CaptureSpan`>> annotation to create a span for the `spanWithAnnotation()` method.
+The span is named `spanName`, is of type `ext`, and subtype `http`.
+
+[source,java]
+----
+@CaptureSpan(value = "spanName", type = "ext", subtype = "http")
+private static void spanWithAnnotation() {
+    // do your thing...
+}
+----
+
+[float]
+==== Combine with the span API
+
+// This content is also used in the `method-api` documentation
+// Ensure any changes made here can also apply there.
+// tag::combine-api-annotations[]
+You can combine annotations with the span API to increase their flexibility.
+Just get the current span on an annotated method and customize the span to your liking.
+
+[source,java]
+----
+@CaptureSpan(value = "spanName", type = "ext", subtype = "http") <1>
+private static void spanWithAnnotation() {
+    Span parent = ElasticApm.currentSpan(); <2>
+    Span span = parent.startSpan(); <3>
+    try {
+        // do your thing...
+    } catch (Exception e) {
+        span.captureException(e);
+        throw e;
+    } finally {
+        span.end();
+    }
+}
+----
+<1> Use `@CaptureSpan` annotation to create a span
+<2> Get the current span
+<3> Start a child span
+// end::combine-api-annotations[]
+
+//********************************************
+[[method-config-based]]
+=== Configuration-based
+
+Use the <<config-trace-methods,`trace_methods`>> configuration option to specify additional methods to instrument.
+You can match methods via wildcards in the package, class or method name, by their modifier (like public),
+by a particular annotation, and more.
+Because you don’t need to modify your source code, this makes it possible to monitor code in 3rd party libraries.
+
+Be careful, it's easy to overuse `trace_methods` by matching too many methods--hurting both runtime and startup performance.
+Use in conjunction with <<config-trace-methods-duration-threshold,`trace_methods_duration_threshold`>> when setting for entire packages
+in order to reduce overhead and avoid having too many spans in the APM app.
+
+[float]
+==== How-to create spans with `trace_methods`
+
+// I don't really understand this.
+// Just gonna put some dance emojis here for now 💃🕺💃🕺💃🕺
+// Move from configuration to here?
```

Comment:
Maybe just link do docs?

Reply:
Yeah, that's probably the best solution for config-based. I don't want to duplicate the content, and we can't easily share content from the configuration reference.